### PR TITLE
refactor: move plugin models to sdk

### PR DESCRIPTION
**Pull request by sbchaos**

### CODE REVIEW

**Review by sravankorumilli — APPROVED** [review]

File: plugin/yaml/plugin.go [hunk 1]
```diff
@@ -100,15 +100,15 @@ func NewPluginSpec(pluginPath string) (*PluginSpec, error) {
 		return nil, err
 	}
 	var plugin PluginSpec
-	if err := yaml.UnmarshalStrict(pluginBytes, &plugin); err != nil {
+	if err := yaml.Unmarshal(pluginBytes, &plugin); err != nil { // TODO: check if strict marshal is required
```
Comment (smarchint) [inline plugin/yaml/plugin.go:103]: why was this removed ? @sbchaos

Comment (sbchaos) [reply plugin/yaml/plugin.go:103]: New plugin model has few items removed, so made it less strict for migration.
